### PR TITLE
Document domain exceptions

### PR DESCRIPTION
**Pull request by rokaszygmantas**

### CODE REVIEW

File: src/content/1.7/development/architecture/domain-exceptions.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
+---
+title: Domain exceptions
+weight: 10
+---
+
+# Domain exceptions
+
+With introduction of domain driven design in PrestaShop, a new way of error handling in the code is seeing the daylight.
+The code in `PrestaShop\PrestaShop\Core\Domain` namespace started to utilize *domain exceptions*.
```
Comment (matks) [inline src/content/1.7/development/architecture/domain-exceptions.md]: 🌈

File: src/content/1.7/development/architecture/domain-exceptions.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    }
+
+    // ...
+}
+```
+
+So now, if we are editing a category and it fails for some reason (e.g. the database is not responding),
+the controller will catch the `CannotEditCategoryException` (which is a child of `CategoryException`) and display a specific error message to the user.
+
+## But there are so many new exception classes. I'm scared!
```
Comment (matks) [inline src/content/1.7/development/architecture/domain-exceptions.md]:
```suggestion
## Why are there so many new exception classes.
```
Maybe a bit more professional 😛

Comment (rokaszygmantas) [reply src/content/1.7/development/architecture/domain-exceptions.md]: oh I thought I reached the pinnacle of poetry :sweat_smile:

File: src/content/1.7/development/architecture/domain-exceptions.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    // ...
+}
+```
+
+So now, if we are editing a category and it fails for some reason (e.g. the database is not responding),
+the controller will catch the `CannotEditCategoryException` (which is a child of `CategoryException`) and display a specific error message to the user.
+
+## Why are there so many new exception classes.
+
+**Don't be!**
```
Comment (matks) [inline src/content/1.7/development/architecture/domain-exceptions.md]:
```suggestion
```

**Comment by matks** [comment]
Thank you @rokaszygmantas

**Pull request merged**